lib/dl_task_processor/dl_task_processor/worker.py

```diff
@@ -22,18 +27,56 @@
 )
 from dl_task_processor.context import BaseContextFabric
 from dl_task_processor.executor import ExecutorFabric
+from dl_task_processor.upstream_worker import Worker
+
 
+LOGGER = logging.getLogger(__name__)
 
 CONTEXT_KEY = "bi_context"
 
 
+class DLArqWorker(Worker):
+    def __init__(self, health_check_record_ttl: SecondsTimedelta, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.health_check_record_ttl = to_seconds(health_check_record_ttl)
+
+    async def record_health(self) -> None:
+        """Copy and paste from the base class, but using health_check_record_ttl"""
+
+        now_ts = time.time()
+        if (now_ts - self._last_health_check) < self.health_check_interval:
+            return
+        self._last_health_check = now_ts
+        pending_tasks = sum(not t.done() for t in self.tasks.values())
+        queued = await self.pool.zcard(self.queue_name)
+        info = (
+            f"{datetime.now():%b-%d %H:%M:%S} j_complete={self.jobs_complete} j_failed={self.jobs_failed} "
+            f"j_retried={self.jobs_retried} j_ongoing={pending_tasks} queued={queued}"
+        )
+        await self.pool.psetex(  # type: ignore[no-untyped-call]
+            self.health_check_key, int(self.health_check_record_ttl * 1000), info.encode()
+        )
+        log_suffix = info[info.index("j_complete=") :]
+        if self._last_health_check_log and log_suffix != self._last_health_check_log:  # TODO?: log health always?
+            LOGGER.info("recording health: %s", info)
+            self._last_health_check_log = log_suffix
+        elif not self._last_health_check_log:
+            self._last_health_check_log = log_suffix
+
+
 @attr.s
 class WorkerSettings:
     # we should not allow forever-fail tasks because it can stop the whole system
     # but (if you really want it) you can provide float('inf')
-    retry_hard_limit: Union[int, float] = attr.ib(default=100)
+    retry_hard_limit: int = attr.ib(default=100)
     job_timeout: int = attr.ib(default=600)  # seconds
-    health_check_interval: int = attr.ib(default=30)
+    health_check_interval: int = attr.ib(default=30)  # how often the HC record is set
+    health_check_record_ttl: int = attr.ib(  # ttl should generally be greater than the HC interval with a margin
```

Comment:
Before the semaphore fix, the default here was `self.job_timeout + self.health_check_interval * 2` (see https://github.com/datalens-tech/datalens-backend/pull/239/commits/165b066347d519a749a0b000eb3305bbf1889303#diff-a2dd8323e818e0401d0172aef8ae41b64d65b0bd48167b67cafc432bf45a4fdfR77), but since the main loop is non-blocking now, we don't necessarily need this big of a margin